concent_api/core/utils.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+import math
+
+from django.conf                    import settings
+
+from golem_messages                 import constants
+from golem_messages                 import message
+from golem_messages.helpers         import maximum_download_time
+from golem_messages.helpers         import subtask_verification_time
+
+
+def calculate_maximum_download_time(size: int, rate: int = constants.DEFAULT_UPLOAD_RATE) -> int:
+    """
+    This function calls `maximum_download_time` helper function from golem messages or Concent custom implementation
+    of it, depending on CUSTOM_PROTOCOL_TIMES setting value.
+    The reason for using custom implementation is because it has hard-coded values and we cannot make it use values
+    from our settings.
+    """
+    assert isinstance(size, int)
+    assert isinstance(rate, int)
+    assert size > 0
+    assert rate > 0
+    assert settings.DOWNLOAD_LEADIN_TIME > 0
+
+    if settings.CUSTOM_PROTOCOL_TIMES:
+        bytes_per_sec = rate << 10
+        download_time = int(
+            datetime.timedelta(
+                seconds=int(math.ceil(size / bytes_per_sec))
+            ).total_seconds()
+        )
+
+        return settings.DOWNLOAD_LEADIN_TIME + download_time
+    else:
+        return int(maximum_download_time(size, settings.MINIMUM_UPLOAD_RATE).total_seconds())
```

Comment:
Please don't hard-code `settings.MINIMUM_UPLOAD_RATE`. You should use the `rate` parameter here.

Reply:
:+1: